From: Joy Latten <joy.latten@canonical.com>
Forwarded: not-needed
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/1884265
Description: ntpq crashes using fips-openssl

Index: ntp-4.2.8p10+dfsg/ntpq/ntpq.c
===================================================================
--- ntp-4.2.8p10+dfsg.orig/ntpq/ntpq.c
+++ ntp-4.2.8p10+dfsg/ntpq/ntpq.c
@@ -3645,8 +3645,14 @@ static void list_md_fn(const EVP_MD *m,
      */
 
     ctx = EVP_MD_CTX_new();
-    EVP_DigestInit(ctx, EVP_get_digestbyname(name));
-    EVP_DigestFinal(ctx, digest, &digest_len);
+    if (EVP_DigestInit(ctx, EVP_get_digestbyname(name)) <= 0) {
+        EVP_MD_CTX_free(ctx);
+        return;
+    }
+    if (EVP_DigestFinal(ctx, digest, &digest_len) <= 0) {
+        EVP_MD_CTX_free(ctx);
+	return;
+    }
     EVP_MD_CTX_free(ctx);
     if (digest_len > (MAX_MAC_LEN - sizeof(keyid_t)))
         return;
